### ISSUE TITLE
jib:build is not reproducible due to directory timestamps not being set

### ISSUE THREAD

**Issue by eisig** [GoogleContainerTools/jib#1648]
**Description of the issue**:
When I build the code in container,  it  will rebuild all the layer every time. 

**Expected behavior**:
 I don’t have to wait for Docker to rebuild your entire Java application - just deploy the layers that changed.
**Steps to reproduce**: 
1. clone https://github.com/GoogleContainerTools/jib.git , cd  to `examples/spring-boot`
2. build with:
```
docker run -v `pwd`:/tmp/build -w /tmp/build -v $HOME/.m2:/root/.m2 maven:3-jdk-8 mvn clean compile jib:build -Djib.from.image=openjdk:8-alpine -Dimage=<myimage>
```
3. `docker pull` the new image.  I have to pull all the 3 layers, include the 14.38MB dependency layer every time .

**Environment**: 
macos 10.14.3, docker Version 2.0.0.3

**Comment by loosebazooka** [comment]
Can you run docker history on the image to see if the digests are actually different?

**Comment by chanseokoh** [comment]
Seems like a critical bug. Timestamps of directories are not set to the epoch+1 sec.

```
examples/spring-boot$ tar tvf target/jib-cache/layers/01*/f8*
drwxr-xr-x 0/0               0 2019-04-19 13:28 app/
drwxr-xr-x 0/0               0 2019-04-19 13:28 app/classes/
drwxr-xr-x 0/0               0 1969-12-31 19:00 app/classes/hello/
-rw-r--r-- 0/0             699 1969-12-31 19:00 app/classes/hello/Application.class
-rw-r--r-- 0/0             594 1969-12-31 19:00 app/classes/hello/HelloController.class
```

We need to track down since when this started to happen.

**Comment by chanseokoh** [comment]
It doesn't have to run inside the container to reproduce.

**Comment by loosebazooka** [comment]
I think #1571 added this regression: https://github.com/GoogleContainerTools/jib/pull/1571/files#diff-9f3ce3d4c2254dbe95c91624e7b434d9R110

**Comment by loosebazooka** [comment]
Moving the time code in the link above doesn't account for parent directories added with `uniqueTarArchiveEntries.add(entry);`

**Comment by chanseokoh** [comment]
Gosh. That explains it. That is in 1.1.0.

**Comment by loosebazooka** [comment]
okay, fix is coming, behavior:
- By default timestamp will be epoch+1.second
- Order matters in layerEntries: to correctly configure a directory it must come before any file that exists in it, otherwise it will get default properties.

**Comment by loosebazooka** [comment]
@eisig thanks again for reporting this. We just released 1.1.2 with the fix.

**Comment by eisig** [comment]
Great 👍
